ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/wholeBodyHardwareControl/AvatarMultiThreadingManager.java

```diff
@@ -209,26 +246,51 @@ else if (!useRealtimeThreads && useMultiThreading)
    public void start()
    {
       if (useRealtimeThreads)
-         masterThread.start();
+      {
+         running = true;
+         ((RealtimeThread) masterThread).start();
+      }
       else
-         threadScheduler.schedule(this::doControl, periodicParameters.getPeriod().asNanoseconds(), TimeUnit.NANOSECONDS);
+      {
+         ((RepeatingTaskThread) masterThread).setFrequencyLimit(Conversions.secondsToHertz(masterThreadDt));
+         ((RepeatingTaskThread) masterThread).startRepeating();
+      }
    }
 
    public void join()
    {
       if (useRealtimeThreads)
-         masterThread.join();
+         ((RealtimeThread) masterThread).join();
    }
 
    public void stop()
    {
+      hardwareCommunicationInterface.stop();
+   }
+
+   public void destroy()
+   {
+      hardwareCommunicationInterface.destroy();
+
+      ThreadTools.sleep(500L);
+
+      threadScheduler.dispose();
+
       if (yoVariableServer != null)
          yoVariableServer.close();
-      threadManager.dispose();
-      if (!useRealtimeThreads)
+
+      running = false;
+
+      if (useRealtimeThreads)
       {
-         threadScheduler.shutdown();
+         ((RealtimeThread) masterThread).join();
       }
+      else
+         ((RepeatingTaskThread) masterThread).stopRepeating();
```

Comment:
this is kind of a preference and/or style guide thing, but if you do the brackets on one, it's preferred to do them on both.